apollo-router/src/services/layers/persisted_queries/mod.rs

```diff
@@ -766,119 +789,208 @@ mod tests {
             response.errors,
             vec![graphql_err_operation_not_in_safelist()]
         );
+        let mut metric_attributes = vec![opentelemetry::KeyValue::new(
+            "persisted_queries.safelist.rejected.unknown".to_string(),
+            true,
+        )];
+        if log_unknown {
+            metric_attributes.push(opentelemetry::KeyValue::new(
+                "persisted_queries.logged".to_string(),
+                true,
+            ));
+        }
+        assert_counter!(
+            "apollo.router.operations.persisted_queries",
+            counter_value,
+            &metric_attributes
+        );
     }
 
     async fn allowed_by_safelist(
         pq_layer: &PersistedQueryLayer,
         query_analysis_layer: &QueryAnalysisLayer,
         body: &str,
+        log_unknown: bool,
+        skip_enforcement: bool,
+        counter_value: u64,
     ) {
         let request_with_analyzed_query =
-            run_first_two_layers(pq_layer, query_analysis_layer, body).await;
+            run_first_two_layers(pq_layer, query_analysis_layer, body, skip_enforcement).await;
 
         pq_layer
             .supergraph_request_with_analyzed_query(request_with_analyzed_query)
             .await
             .ok()
             .expect("pq layer second hook returned error response instead of returning a request");
-    }
 
-    #[tokio::test(flavor = "multi_thread")]
-    async fn pq_layer_freeform_graphql_with_safelist() {
-        let manifest = HashMap::from([(
-            FullPersistedQueryOperationId {
-              operation_id: "valid-syntax".to_string(),
-              client_name: None,
-            },
-            "fragment A on Query { me { id } }    query SomeOp { ...A ...B }    fragment,,, B on Query{me{name,username}  } # yeah"
-                .to_string(),
-        ), (
-            FullPersistedQueryOperationId {
-              operation_id: "invalid-syntax".to_string(),
-              client_name: None,
-            },
-            "}}}".to_string()),
-        ]);
+        let mut metric_attributes = vec![];
+        if skip_enforcement {
+            metric_attributes.push(opentelemetry::KeyValue::new(
+                "persisted_queries.safelist.enforcement_skipped".to_string(),
+                true,
+            ));
+            if log_unknown {
+                metric_attributes.push(opentelemetry::KeyValue::new(
+                    "persisted_queries.logged".to_string(),
+                    true,
+                ));
+            }
+        }
 
-        let (_mock_guard, uplink_config) = mock_pq_uplink(&manifest).await;
+        assert_counter!(
```

Comment:
Note that I wrote and tested all the assert_counter calls on a clean checkout before I rewrote the u64_counter calls to use the attributes vector because I was a bit paranoid that I would do something accidental like change bool true to string "true".